### PR TITLE
Document `null` taint

### PR DESCRIPTION
**Pull request by pkesseli**
Add pulse test documenting `null` taint issue.

### CODE REVIEW

File: infer/tests/codetoanalyze/java/pulse/taint/NullPointer.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+package codetoanalyze.java.pulse;
+
+class NullPointer {
+
+    void source(StringBuilder request) {
+    }
+
+    void foo() {
```
Comment (geralt-encore) [inline infer/tests/codetoanalyze/java/pulse/taint/NullPointer.java:15]: Could you rename it to  something like`nullPointerTaintedFalsePositive` to make it more explicit?
Also, it would probably be nice to add a test case for String literals as well.

Comment (pkesseli) [reply infer/tests/codetoanalyze/java/pulse/taint/NullPointer.java]: I renamed the methods and added an example for string literals.

**Review by geralt-encore — APPROVED** [review]
Thanks, I'll merge it as soon as CLA is sorted out
